File: be_codes/KSDenemeDotnet8/Service/impl/CategoryServiceImpl.cs
using KSDenemeDotnet8.Entity;
using KSDenemeDotnet8.Repository;

namespace KSDenemeDotnet8.Service.impl
{
    public class CategoryServiceImpl
    {
        private readonly CategoryRepository rep
            ;

        public CategoryServiceImpl(CategoryRepository _rep)
        {
            rep = _rep;
        }
        public List<Category> getAllCategory()
        {
            return rep.FindAll();
        }

        public Category getCategory(Guid id)
        {
            return rep.FindById(id);
        }

        public void save(Category category)
        {
             rep.Save(category);
        }
    }
}

File: Users/src/main/java/com/example/users/business/concretes/CategoryServiceImpl.java
package com.example.users.business.concretes;

import com.example.users.business.abstracts.CategoryService;
import com.example.users.dataAccess.CategoryRepository;
import com.example.users.entities.concretes.Category;
import com.example.users.model.dto.CategoryDto;
import com.example.users.model.mapper.CategoryMapper;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class CategoryServiceImpl implements CategoryService {
    private final CategoryRepository categoryRepository;

    @Override
    public CategoryDto save(CategoryDto categoryDto) {
        return CategoryMapper.INSTANCE.entitytoDto(categoryRepository.save(CategoryMapper.INSTANCE.dtotoEntity(categoryDto)));
    }

    @Override
    public Set<CategoryDto> getAll() {
        List<Category> allCategories = categoryRepository.findAll();
        Set<CategoryDto> categories = allCategories.stream()
                .map(category -> CategoryMapper.INSTANCE.entitytoDto(category)).collect(Collectors.toSet());
        return categories;
    }
}
